.ci.sh

```diff
@@ -8,27 +8,93 @@ set -eo pipefail  # exit if non-zero error is encountered (even in a pipeline)
 set -u            # exit if unset variables used
 shopt -s failglob # error if a glob doesn't find any files, instead of remaining unexpanded
 
+export PIP_PROGRESS_BAR=off # disable pip's progress bar for the duration of CI
+
 echo Installing SCT
-PIP_PROGRESS_BAR=off ./install_sct -y
+# NB: we only force in-place (-i) installs to avoid pytest running from the source
+#     instead of the installed folder, where the extra detection models are.
+#     Further explanation at https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure
+#     TO BE REMOVED during https://github.com/neuropoly/spinalcordtoolbox/issues/3140.
+./install_sct -iy
+set +euo pipefail; . ~/.bash_profile; set -euo pipefail;  # load sct
 
 echo *** CHECK PATH ***
-ls -lA bin  # Make sure all binaries and aliases are there
-source python/etc/profile.d/conda.sh  # to be able to call conda
+# Make sure all binaries and aliases are there
+for tool in \
```

Comment:
i'm sure you already thought about it and there is probably no better workaround than your suggestion, but listing all functions seems prone to "forgetting adding one if in the future someone adds a new script". Isn't there a more automated approach, e.g. doing an ls on `spinalcordtoolbox/scripts` folder?

Reply:
But doing an `ls` doesn't actually test anything.  I've seen plenty of CI runs where that section just comes out empty, but you'd never know if unless you dug. `command -v` actually makes sure the commands are callable. Maybe we should just remove this check entirely since it's not helping?

I agree that maintaining the list is tricky. I'm open to suggestions? Maybe @Drulex or @cfhammill or @Drulex have some? Can I tag the newer people too for their ideas too?

What if we set up an ad-hoc [snapshot test](https://jestjs.io/docs/en/snapshot-testing): we have this script log the output of `ls` to a file if it doesn't exist, then compare the contents of that file to `ls` again, and if they differ, crash. Updating it would be:

```
rm ~/.snapshots/bin.ls.txt
/.ci.sh
git add -u ~/.snapshots
git commit -m "Update snapshot test because [ I added a new script, I deleted a script, etc ]"
```

Reply:
> I agree that maintaining the list is tricky. I'm open to suggestions? Maybe @Drulex or @cfhammill or @Drulex have some? Can I tag the newer people too for their ideas too?

@joshuacwnewton @ahill187 @gisellemartel @charjl

Reply:
> Maybe we should just remove this check entirely since it's not helping?

This still seems like a useful check, but maybe not necessary for the scope of this PR? I could see removing it for now and then opening an issue to add it separately.

Either way, if we were to add a check like this, perhaps a better place is `sct_check_dependencies`, because that's effectively our "validate installation" command. (We often ask users to run this when they're reporting a bug.) See these related checks:

```bash
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]  # FYI: This calls `isct_test_ants`
Check PropSeg compatibility with OS ................[OK]  # FYI: This calls `isct_propseg`
```

Reply:
As for maintaining a list of binaries... we're actually already manually maintaining a list like this :sweat_smile::

https://github.com/neuropoly/spinalcordtoolbox/blob/3dfd4f81f9e4364dab4d94894ea687a625dbdea5/setup.py#L58-L112

However, this is just the CLI scripts and not the other binaries. But still, I could see having a single list of binaries that both `setup.py` pulls from for installation, then `.ci.sh` (or `sct_check_dependencies`) pulls from for validation.

Reply:
I'm just going to get rid of this part: https://github.com/neuropoly/spinalcordtoolbox/pull/3125#discussion_r553114359.

Reply:
e92285f31c4f7fd1fac6a6c7f2a6dd6c323cb5b3